RZAssemblage/RZArrayAssemblage.m

```diff
@@ -0,0 +1,170 @@
+//
+//  RZArrayAssemblage.m
+//  RZAssemblage
+//
+//  Created by Brian King on 3/21/15.
+//  Copyright (c) 2015 Raizlabs. All rights reserved.
+//
+
+#import "RZArrayAssemblage.h"
+#import "RZAssemblage+Private.h"
+
+NSString *const RZAssemblageUpdateKey = @"RZAssemblageUpdateKey";
+static char RZAssemblageUpdateContext;
+
+@implementation RZArrayAssemblage
+
++ (BOOL)shouldObserveContents
+{
+    return YES;
+}
+
+- (instancetype)initWithArray:(NSArray *)array representingObject:(id)representingObject;
+{
+    self = [super init];
+    if ( self ) {
+        self.representedObject = representingObject;
+        _childrenStorage = [array isKindOfClass:[NSMutableArray class]] ? array : [array mutableCopy];
+        if ( self.class.shouldObserveContents ) {
+            [[_childrenStorage copy] enumerateObjectsUsingBlock:^(id object, NSUInteger idx, BOOL *stop) {
+                [self addMonitorsForObject:object];
```

Comment:
You do the `copy` here to safely enumerate, but in reality if one of the objects in the original array is removed during enumeration, you'll have added observers (and have no way to remove them) for an object that may then get deallocated. If you're worried enough to do the `copy`, seems like a better solution to `@synchronized (array)`